examples/vector_databases/analyticdb/QA_with_Langchain_AnalyticDB_and_OpenAI.ipynb

```diff
@@ -0,0 +1,442 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Question Answering with Langchain, AnalyticDB and OpenAI\n",
+    "This notebook presents how to implement a Question Answering system with Langchain, AnalyticDB as a knowledge based and OpenAI embeddings. If you are not familiar with AnalyticDB, it’s better to check out the Getting_started_with_AnalyticDB_and_OpenAI.ipynb notebook.\n",
+    "This notebook presents an end-to-end process of:\n",
+    "Calculating the embeddings with OpenAI API.\n",
+    "Storing the embeddings in an AnalyticDB instance to build a knowledge base.\n",
+    "Converting raw text query to an embedding with OpenAI API.\n",
+    "Using AnalyticDB to perform the nearest neighbour search in the created collection to find some context.\n",
+    "Asking LLM to find the answer in a given context.\n",
+    "All the steps will be simplified to calling some corresponding Langchain methods."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Prerequisites\n",
+    "For the purposes of this exercise we need to prepare a couple of things:\n",
+    "[AnalyticDB cloud instance](https://www.alibabacloud.com/help/en/analyticdb-for-postgresql/latest/product-introduction-overview).\n",
+    "Langchain as a framework.\n",
+    "An OpenAI API key."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Install requirements\n",
+    "This notebook obviously requires the `openai`, `langchain`, `psycopg2cffi` and `tiktoken`"
```

Comment:
optional: explain what each of these is for, add links (I wasn't sure what psycopg2ffi was for and had to look it up)